picard/ui/options/scripting.py

```diff
@@ -103,18 +120,113 @@ class ScriptingOptionsPage(OptionsPage):
     ACTIVE = True
 
     options = [
-        config.BoolOption("setting", "enable_tagger_script", False),
-        config.TextOption("setting", "tagger_script", ""),
+        config.BoolOption("setting", "enable_tagger_scripts", False),
+        config.ListOption("setting", "list_of_scripts", []),
     ]
 
+    SCRIPT_DEFAULT_NAME = N_("My Script")
+
     def __init__(self, parent=None):
         super(ScriptingOptionsPage, self).__init__(parent)
         self.ui = Ui_ScriptingOptionsPage()
         self.ui.setupUi(self)
         self.highlighter = TaggerScriptSyntaxHighlighter(self.ui.tagger_script.document())
-        self.ui.tagger_script.textChanged.connect(self.live_checker)
+        self.ui.tagger_script.textChanged.connect(self.live_update_and_check)
+        self.ui.add_script.clicked.connect(self.add_to_lscript)
+        self.ui.remove_script.clicked.connect(self.remove_from_lscript)
+        self.ui.up_script.clicked.connect(lambda: self.move_script(1))
+        self.ui.down_script.clicked.connect(lambda: self.move_script(-1))
+        self.ui.script_list.itemSelectionChanged.connect(self.script_selected)
+        self.ui.script_list.itemChanged.connect(self.script_attr_changed)
+        self.ui.tagger_script.setEnabled(False)
+        self.listitem_to_scriptitem = {}
+        self.list_of_scripts = []
+
+    def script_attr_changed(self, item):
+        item.setSelected(True)
+        script = self.listitem_to_scriptitem[item]
+        if item.checkState():
+            script.enabled = True
+        else:
+            script.enabled = False
+        script.name = item.text()
+        self.list_of_scripts[script.pos] = script.get_all()
+
+    def script_selected(self):
+        items = self.ui.script_list.selectedItems()
+        if items:
+            self.ui.tagger_script.setEnabled(True)
+            script = self.listitem_to_scriptitem[items[0]]
+            self.ui.tagger_script.setText(script.text_item)
+
+    def add_to_lscript(self):
+        count = self.ui.script_list.count()
+        script = ScriptItem(position=count, title=_(self.SCRIPT_DEFAULT_NAME)+" "+str(count+1))
```

Comment:
i18n issue here, don't concatenate numbers to strings this way.
```python
DEFAULT_NUMBERED_SCRIPT_NAME = N_("My script %d")
title = _(DEFAULT_NUMBERED_SCRIPT_NAME) % (count + 1) 
```

Main reason is that some languages may place number before the string '1 string' vs 'string 1' and handling of plural forms depends on languages.
So concatenating numbers and strings is usually a bad idea when it comes to translations.